discretize/CylMesh.py

```diff
@@ -146,72 +328,715 @@ def vectorCCx(self):
+        """
+        dictionary of the indices of the hanging x-edges (keys) and a list
+        of indices that the eliminated faces map to (if applicable)
+        """
+        if getattr(self, '_hangingExDict', None) is None:
+            deflate_y = np.zeros(self._ntNy, dtype=bool)
+            deflate_y[0] = True
+            deflateEx = np.nonzero(np.kron(
+                np.ones(self._ntNz, dtype=bool),
+                np.kron(
+                    deflate_y,
+                    np.ones(self.nCx, dtype=bool)
+                )
+            ))[0].tolist()
+            self._hangingExDict = dict(zip(
+                np.nonzero(self._ishangingEx)[0].tolist(), deflateEx
+            ))
+        return self._hangingExDict
+
+    @property
+    def _ishangingEy(self):
+        """
+        bool vector indicating if a y-edge is hanging or not
+        """
+        if getattr(self, '_ishangingEyBool', None) is None:
+            hang_x = np.zeros(self._ntNx, dtype=bool)
+            hang_x[0] = True
+            self._ishangingEyBool = np.kron(
+                np.ones(self._ntNz, dtype=bool),
+                np.kron(
+                    np.ones(self.nCy, dtype=bool),
+                    hang_x
+                )
+            )
+        return self._ishangingEyBool
+
+    @property
+    def _hangingEy(self):
+        """
+        dictionary of the indices of the hanging y-edges (keys) and a list
+        of indices that the eliminated faces map to (if applicable)
+        """
+        if getattr(self, '_hangingEyDict', None) is None:
+            self._hangingEyDict = dict(zip(
+                np.nonzero(self._ishangingEy)[0].tolist(),
+                [None]*len(self._ishangingEyBool))
+            )
+        return self._hangingEyDict
+
+    @property
+    def _axis_of_symmetry_Ez(self):
+        """
+        bool vector indicating if a z-edge is along the axis of symmetry or not
+        """
+        if getattr(self, '_axis_of_symmetry_EzBool', None) is None:
+            axis_x = np.zeros(self._ntNx, dtype=bool)
+            axis_x[0] = True
+
+            axis_y = np.zeros(self._ntNy, dtype=bool)
+            axis_y[0] = True
+            self._axis_of_symmetry_EzBool = np.kron(
+                np.ones(self.nCz, dtype=bool),
+                np.kron(
+                    axis_y,
+                    axis_x
+                )
+            )
+        return self._axis_of_symmetry_EzBool
+
+    @property
+    def _ishangingEz(self):
+        """
+        bool vector indicating if a z-edge is hanging or not
+        """
+        if getattr(self, '_ishangingEzBool', None) is None:
+            if self.isSymmetric:
+                self._ishangingEzBool = np.ones(self._ntEz, dtype=bool)
+            else:
+                hang_x = np.zeros(self._ntNx, dtype=bool)
+                hang_x[0] = True
+
+                hang_y = np.zeros(self._ntNy, dtype=bool)
+                hang_y[-1] = True
+
+                hangingEz = np.kron(
```

Comment:
I think this is exactly the same?
```
hanging = np.zeros(self._vntEz, dtype=bool)
hanging[0, :, :] = True  # The first z-edges are the same
hanging[:, -1, :] = True  # All the z-edges wrap
hanging[0, 0, :] = False  # Not the axis of symmetry
return mkvc(hanging)
```

Reply:
It is, but unfortunately working in 3D and then flattening is way less efficient 
![image](https://user-images.githubusercontent.com/6361812/38274501-cad20936-3743-11e8-9cff-4a390dde7ee6.png)